File: src/main/scala/dev/tlang/tlang/ast/common/ClassType.scala
package dev.tlang.tlang.ast.common

import tlang.core
import tlang.core.Type

case class ClassType(clazz: Class[_]) extends Type {
  override def getType: core.String = new core.String(clazz.getPackageName + "/" + clazz.getSimpleName)

  override def getSimpleType: core.String = new core.String(clazz.getSimpleName)

  override def getPkg: core.String = new core.String(clazz.getPackageName)
}

File: src/main/java/tlang/internal/ClassType.java
package tlang.internal;

import tlang.core.String;
import tlang.core.Type;

public class ClassType implements Type {

    private final Class<?> clazz;

    public ClassType(Class<?> clazz) {
        this.clazz = clazz;
    }

    @Override
    public String getType() {
        return new String((clazz.getPackageName() + "/" + clazz.getSimpleName()));
    }

    @Override
    public String getSimpleType() {
        return new String(clazz.getSimpleName());
    }

    @Override
    public String getPkg() {
        return new String(clazz.getPackageName());
    }

    public static Type of(Class<?> clazz) {
        return new ClassType(clazz);
    }
}
